### DISCUSSION TITLE
Search Engine

### DISCUSSION THREAD

**Discussion by emilianoyaryura** [supabase/supabase#14301] [Questions]
Hi, I have a question about how Supabase charges and if this workaround I will show is somehow possible.

I'm building a side project and I am in a heavy development stage, therefore I can't expend that much money, neither I can in a future, because my idea is not that profitable.

If I want to have a search engine so the user can look through all posts and other users, is this approach possible? I would be calling this API every 500ms so the user can see results while he writes in the input.

My main concern is anything breaking because of the number of possible requests

```
import { NextApiRequest, NextApiResponse } from "next";
import { tags } from "../../../lib/constants/tags";
import { supabase } from "../../../lib/supabase-client";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  const { searchQuery } = req.query;

  // Search in users
  const { data: usersData, error: usersError } = await supabase
    .from("user")
    .select("id, name, username")
    .filter("username", "ilike", `%${searchQuery}%`)
    .or(`name.ilike.%${searchQuery}%`);

  if (usersError) return res.status(500).json({ error: usersError.message });

  // Search in posts
  const { data: postsData, error: postsError } = await supabase
    .from("post")
    .select("id, title, slug")
    .filter("title", "ilike", `%${searchQuery}%`);

  if (postsError) return res.status(500).json({ error: postsError.message });

  // Local tags search
  const tagsData = tags.filter(
    (tag) =>
      tag.en.toLowerCase().includes(String(searchQuery).toLowerCase()) ||
      tag.es.toLowerCase().includes(String(searchQuery).toLowerCase())
  );

  return res
    .status(200)
    .json({ users: usersData, posts: postsData, tags: tagsData });
}

```

**Comment by GaryAustin1** [comment]
I have a filter for searching titles of notes and start it after 2 characters.  I then do an ilike query with a limit of 10 and also ask for a count of matches.  I only return titles and id in this query.  I display the results and then indicate there are extra matches if count is greater than the limit.  

As the user types I rerun the query.  

If the user wants to see the actual note they click on the title and then get all the note info so it is not being downloaded constantly as the filter query runs.

Not sure that is a helpful idea or not, but I don't see you having any limit in your queries which you probably want and have the count so you know there are more and can show that.
